hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -880,137 +797,143 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
-            ).result_price
-        except EnvironmentError:
-            top_ask_price = maker_order_book.c_get_price(True)
 
         # Calculate the next price from the top, and the order size limit.
         if is_bid:
+
             price_quantum = maker_market.c_get_order_price_quantum(
                 market_pair.maker.trading_pair,
                 top_bid_price
             )
-            next_price = (round(Decimal(top_bid_price) / price_quantum) + 1) * price_quantum
+            price_above_bid = (round(Decimal(top_bid_price) / price_quantum) + 1) * price_quantum
 
-            # Calculate the order size limit from maker and taker market balances.
-            maker_balance_size_limit = (maker_market.c_get_available_balance(market_pair.maker.quote_asset) /
-                                        float(next_price))
-            taker_balance_size_limit = (taker_market.c_get_available_balance(market_pair.taker.base_asset) *
-                                        self._order_size_taker_balance_factor)
+            try:
+                taker_price = taker_order_book.c_get_vwap_for_volume(False,
+                                                                     size).result_price
+            except ZeroDivisionError:
+                return None
 
-            # Convert the proposed maker order price to the equivalent price on the taker market.
-            adjusted_taker_price = (self._exchange_rate_conversion.adjust_token_rate(
-                market_pair.maker.quote_asset,
-                float(next_price)
-            ) / self._exchange_rate_conversion.adjust_token_rate(
-                market_pair.taker.quote_asset,
-                1.0
-            ))
-
-            # Calculate the order size limit from the minimal profitable hedge on the taker market.
-            profitable_hedge_price = adjusted_taker_price * (1 + self._min_profitability)
-            taker_order_book_size_limit = (
-                taker_order_book.c_get_volume_for_price(False, profitable_hedge_price).result_volume *
-                self._order_size_taker_volume_factor
-            )
+            # If quote assets are not same, convert them from taker's quote asset to maker's quote asset
+            if market_pair.maker.quote_asset != market_pair.taker.quote_asset:
+                taker_price *= self._exchange_rate_conversion.convert_token_value(1,
+                                                                                  market_pair.taker.quote_asset,
+                                                                                  market_pair.maker.quote_asset)
+
+            # you are buying on the maker market and selling on the taker market
+            maker_price = taker_price / (1 + self._min_profitability)
+
+            if self._adjust_orders_enabled:
+                # If your bid is higher than highest bid price, reduce it to one tick above the top bid price
+                if maker_price > price_above_bid:
```

Comment:
`maker_price = min(maker_price, price_above_bid)` would be simpler and easier to read.